app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -2,148 +2,197 @@
 
 class DeleteObjectHierarchyWorker < ApplicationJob
 
-  # TODO: Rails 5 --> discard_on ActiveJob::DeserializationError
-  # No need of ActiveRecord::RecordNotFound because that can only happen in the callbacks and those callbacks don't use this rescue_from but its own rescue
-  rescue_from(ActiveJob::DeserializationError) do |exception|
-    Rails.logger.info "DeleteObjectHierarchyWorker#perform raised #{exception.class} with message #{exception.message}"
-  end
+  WORK_TIME_LIMIT_SECONDS = 5
+  ASSOCIATION_RE = /Association-(?<klass>[:\w]+)-(?<id>\d+):(?<association>\w+)/
 
-  queue_as :deletion
+  class DoNotRetryError < RuntimeError; end
 
-  before_perform do |job|
-    @object, workers_hierarchy, @background_destroy_method = job.arguments
-    id = "Hierarchy-#{object.class.name}-#{object.id}"
-    @caller_worker_hierarchy = Array(workers_hierarchy) + [id]
-    info "Starting #{job.class}#perform with the hierarchy of workers: #{caller_worker_hierarchy}"
+  rescue_from(DoNotRetryError) do |exception|
+    # report error and skip retries
+    System::ErrorReporting.report_error(exception)
   end
 
-  after_perform do |job|
-    info "Finished #{job.class}#perform with the hierarchy of workers: #{caller_worker_hierarchy}"
+  # we need this only for compatibility to process already enqueued jobs after upgrade
+  rescue_from(ActiveJob::DeserializationError) do |exception|
+    System::ErrorReporting.report_error(exception)
   end
 
-  def perform(_object, _caller_worker_hierarchy = [], _background_destroy_method = 'destroy')
-    build_batch
-  end
+  queue_as :deletion
 
-  def on_success(_, options)
-    on_finish('on_success', options)
-  end
+  # until_executed seems to rely on #after_perform which is skipped on failure so not sure whether retries will work.
+  # additionally if we want to reschedule ourselves in #after_perform, it is not nice to rely on flaky order
+  # see https://github.com/3scale/activejob-uniqueness/blob/main/lib/active_job/uniqueness/strategies/until_executed.rb
```

Comment:
```suggestion
  # see https://github.com/veeqo/activejob-uniqueness/blob/v0.2.5/lib/active_job/uniqueness/strategies/until_executed.rb
```

Apparently, we do not rely on our fork anymore, so probably this repo under 3scale needs to go away at some point.
So, I'd say we better link the original repo, and I've added the current version - in case implementation changes in later versions.